newIDE/app/src/MainFrame/index.js

```diff
@@ -1665,6 +1670,98 @@ const MainFrame = (props: Props) => {
       message: 'Update available',
     });
 
+  // Very temporary keyboard shortcut for command palette
+  React.useEffect(() => {
+    const handler = (e: KeyboardEvent) => {
+      const body = document.body;
+      const activeEl = document.activeElement;
+      const mainFrame = document.querySelector('div.main-frame');
+      const isBody = activeEl === body;
+      const isInMainframe = mainFrame && mainFrame.contains(activeEl);
+      if (!isBody && !isInMainframe) return;
+      if ((e.ctrlKey || e.metaKey) && e.code === 'KeyP')
+        openCommandPalette(true);
+    };
+    document.addEventListener('keyup', handler);
+    return () => document.removeEventListener('keyup', handler);
+  }, []);
```

Comment:
Isn't there a dependency missing (at least openCommandPalette)?

Reply:
Actually no, nothing's missing. No warnings either :)
I was surprised about it too, when omitting `openCommandPalette` did not raise a warning - I guess the ESLint hooks rule has recently started to consider that the "setSomething" functions coming directly from `React.useState` don't really change themselves.

Reply:
That's great then 👍